plugin/src/App/init.lua

```diff
@@ -59,15 +63,57 @@ function App:startSession(host, port, sessionOptions)
 				appStatus = AppStatus.Connecting,
 			})
 		elseif status == ServeSession.Status.Connected then
+			if expectedSessionId and details.sessionId ~= expectedSessionId then
+				ServeSession:stop()
+			end
+
 			local address = ("%s:%s"):format(host, port)
+
+			if RunService:IsClient() and RunService:IsServer() and not RunService:IsRunning() then
+				local connectionId = HttpService:GenerateGUID()
+				local activeConnections = settings:get("activeConnections", true)
+				table.insert(activeConnections, {
+					connectionId = connectionId,
+					sessionId = details.sessionId,
+					host = host,
+					port = port
+				})
+
+				if #activeConnections > 15 then
+					table.remove(activeConnections, 1)
+				end
+
+				settings:set("activeConnections", activeConnections)
```

Comment:
I don't understand why we have this array and write it to `settings.json`.

You're creating a GUID to key the host/port/sessionId within this array, then sharing the GUID via the Attribute, then linearly searching the array to find the host/port/sessionId with this GUID when play solo.
(Why are do a linear search? You can turn this array into a dict using the GUID as a key.)

I think this array and disc write is totally avoidable though, unless there's something I'm missing.
You can write the connection info to an easily parse-able string in the Attribute, so that the server window of the test can grab the host/port/sessionId without having to search through other connections from other projects in other windows.

IE:
```Lua
-- When connecting
workspace:SetAttribute("__RojoConnectionInfo", host .. "|" .. port .. "|" .. details.sessionId)

-- When in play solo
local connectionInfo = workspace:GetAttribute("__RojoConnectionInfo")
if not connectionInfo then return end
local host, port, sessionId = table.unpack(string.split(connectionInfo, "|"))
```

Reply:
I agree that the disc write isn’t the way to go. It felt shaky even when I implemented it.

 As for the rest of your questions— I don’t have good answers for those, mostly because I can’t remember and I’d need to read back through the whole PR, which I’m not really interested in doing right now. (nor am I interested in continually addressing those issues in this PR)

I think it would be better for you to restart this feature from scratch

Reply:
Yup, I opened #840 for that. Thanks for getting this started though!